x-pack/plugins/security_solution/public/detection_engine/rule_management_ui/components/rules_table/rules_table/use_sync_rules_table_saved_state.test.tsx

```diff
@@ -0,0 +1,296 @@
+      services: { sessionStorage: { set: setStorage, remove: removeStorage } },
+    });
+  });
+
+  it('clears the default state when there is nothing to sync', () => {
+    (useRulesTableContext as jest.Mock).mockReturnValue({ state: defaultState });
+
+    renderHook(() => useSyncRulesTableSavedState());
+
+    expect(updateUrlParam).toHaveBeenCalledWith(null);
+    expect(setStorage).not.toHaveBeenCalled();
+    expect(removeStorage).toHaveBeenCalledWith(RULES_TABLE_STATE_STORAGE_KEY);
+  });
+
+  it('syncs both the url and the storage', () => {
+    const state = {
+      filterOptions: {
+        filter: 'test',
+        tags: ['test'],
+        showCustomRules: true,
+        showElasticRules: false,
+      },
+      sortingOptions: {
+        field: 'name',
+        order: 'asc',
+      },
+      pagination: {
+        page: 3,
+        perPage: 10,
+        total: 100,
+      },
+    };
+    const expected = {
+      searchTerm: 'test',
+      showCustomRules: true,
+      tags: ['test'],
+      sort: {
+        field: 'name',
+        order: 'asc',
+      },
+      page: 3,
+      perPage: 10,
+    };
+
+    (useRulesTableContext as jest.Mock).mockReturnValue({
+      state,
+    });
+
+    renderHook(() => useSyncRulesTableSavedState());
+
+    expect(updateUrlParam).toHaveBeenCalledWith(expected);
+    expect(setStorage).toHaveBeenCalledWith(RULES_TABLE_STATE_STORAGE_KEY, expected);
+  });
+
+  describe('with the url', () => {
+    it('syncs only the search term', () => {
+      expectUrlSync(
+        { ...defaultState, filterOptions: { ...INITIAL_FILTER_OPTIONS, filter: 'test' } },
+        { searchTerm: 'test' }
+      );
+    });
+
+    it('syncs only the show elastic rules filter', () => {
+      expectUrlSync(
+        {
+          ...defaultState,
+          filterOptions: { ...defaultState.filterOptions, showElasticRules: true },
+        },
+        { showCustomRules: false }
+      );
+    });
+
+    it('syncs only the show custom rules filter', () => {
+      expectUrlSync(
+        {
+          ...defaultState,
+          filterOptions: { ...defaultState.filterOptions, showCustomRules: true },
+        },
+        { showCustomRules: true }
+      );
+    });
+
+    it('syncs only the tags', () => {
+      expectUrlSync(
+        {
+          ...defaultState,
+          filterOptions: { ...defaultState.filterOptions, tags: ['test'] },
+        },
+        { tags: ['test'] }
+      );
+    });
+
+    it('syncs only the sorting field', () => {
+      expectUrlSync(
+        {
+          ...defaultState,
+          sortingOptions: { ...defaultState.sortingOptions, field: 'name' },
+        },
+        { sort: { field: 'name' } }
+      );
+    });
+
+    it('syncs only the sorting order', () => {
+      expectUrlSync(
+        {
+          ...defaultState,
+          sortingOptions: { ...defaultState.sortingOptions, order: 'asc' },
+        },
+        { sort: { order: 'asc' } }
+      );
+    });
```

Comment:
I find it quite difficult to understand what's being tested here. Could you make these test assertions explicit?